### PR TITLE
lock input data when submitting. Fix #7592

### LINKED ISSUE #7592: lock input data when task is submitted
- [x] do this in DBSDataDiscovery
 - [ ] use this chance to look at "clean up pylint complaints"
 - [ ] do we do it for RucioDataDiscovery as well ?

### PR DESCRIPTION
**Pull request by belforte**
still need to change Rucio account,. WIll do after Monday when Rahul is available.

other improvements listed in #7592 will be moved to nee issue.
So that we can deploy this and see how it works.
And code reformatting will be a NOOP functionality-wise

### CODE REVIEW

**Comment by belforte** [comment]
Sorry for long tail of small changes. This is good to go as far as I am concerned.
Please review.

File: src/python/TaskWorker/Actions/DBSDataDiscovery.py [hunk 1]
```diff
@@ -411,8 +412,75 @@ def executeInternal(self, *args, **kwargs):
...
+            # recalling a full DBS dataset, simple the container already exists
+            myScope = 'cms'
+            dbsDatasetName = dataToLock
+            containerDid = {'scope': myScope, 'name': dbsDatasetName}
+        elif isinstance(dataToLock, list):
+            # need to prepare ad hoc container. Can not reuse a possibly existing one
+            # for this dataset which may have had a different content
+            # Therefore we stick with naming the container after the task name
+            myScope = 'user.crab_tape_recall'  # do not mess with cms scope
+            containerName = '/TapeRecall/%s/USER' % self.taskName.replace(':', '.')
```
Comment (novicecpp) [inline src/python/TaskWorker/Actions/DBSDataDiscovery.py:461]: Is this your intention to use the scope/name same as tape recall?

**Review by novicecpp — COMMENTED** [review]
Please see the inline comment.

File: src/python/TaskWorker/Actions/DBSDataDiscovery.py (hunk 1 above)
Comment (belforte) [reply src/python/TaskWorker/Actions/DBSDataDiscovery.py:461]: well spotted. thanks. I will use new account `crab_input` here as well.

Comment (belforte) [reply src/python/TaskWorker/Actions/DBSDataDiscovery.py]: fixed in https://github.com/dmwm/CRABServer/pull/7732/commits/e04e165ea3e1a31d8c2190431b596d278b408815

**Review by novicecpp — APPROVED** [review]
LGTM

**Comment by belforte** [comment]
thanks Wa!

**Pull request merged**